### PR TITLE
Add per-app user config

### PR DESCRIPTION
**Pull request by tfussell**
This PR:

- Adds an optional userconfig secret and configmap to each app based on provided values. Immediately for https://github.com/giantswarm/roadmap/issues/682 but also enables more config if needed in the future.
- Moves app definitions entirely into `values.yaml` for increased configurability.

Example values:
```yaml
clusterName: thom3
organization: multi-project
managementCluster: gamma

userConfig:
  netExporter:
    configMap:
      values: |
        dns:
          service: kube-dns
```

I believe this subsumes https://github.com/giantswarm/default-apps-openstack/pull/18 (or at least the config would be set by `kubectl-gs` instead of in this repo).

### Testing

- [x] Fresh install works.
- [x] Upgrade from previous version works.

### Checklist

- [x] Update changelog in `CHANGELOG.md`.
- [x] Make sure `values.yaml` is valid.

### CODE REVIEW

File: helm/default-apps-openstack/templates/apps.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+metadata:
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ $clusterName }}-user-values
+  namespace: {{ $namespace }}
+data:
+  {{- .configMap | toYaml | toString | nindent 2 }}
+{{- end }}
+---
+{{- if .secret }}
```
Comment (kopiczko) [inline helm/default-apps-openstack/templates/apps.yaml:70]:
```suggestion
{{- if .secret }}
---
```

File: helm/default-apps-openstack/values.yaml [hunk 2]
```diff
@@ -1,2 +1,74 @@
-clusterName: ""
-organization: ""
+clusterName: "test"
+organization: "org"
+
+apps:
+  # calico
+  - appName: calico
```
[the remark this answers is not in the document]
Comment (tfussell) [reply helm/default-apps-openstack/values.yaml:6]: I went back and forth between a map and an array here. Good point about overriding values, I'll change it.

File: helm/default-apps-openstack/values.yaml [hunk 3]
```diff
@@ -1,3 +1,75 @@
...
+  kubeStateMetrics:
+    appName: kube-state-metrics
+    chartName: kube-state-metrics-app
+    catalog: default
+    clusterValues:
+      configMap: true
+      secret: false
+    forceUpgrade: true
+    namespace: kube-system
+    version: 1.5.1
```
Comment (tfussell) [inline helm/default-apps-openstack/values.yaml:45]:
```suggestion
    version: 1.7.0
```

File: helm/default-apps-openstack/values.yaml [hunk 4]
```diff
@@ -1,3 +1,75 @@
...
+  metricsServer:
+    appName: metrics-server
+    chartName: metrics-server-app
+    catalog: default
+    clusterValues:
+      configMap: true
+      secret: false
+    forceUpgrade: true
+    namespace: kube-system
+    version: 1.5.0
```
Comment (tfussell) [inline helm/default-apps-openstack/values.yaml:55]:
```suggestion
    version: 1.6.0
```

File: helm/default-apps-openstack/values.yaml [hunk 5]
```diff
@@ -1,3 +1,75 @@
...
+  nodeExporter:
+    appName: node-exporter
+    chartName: node-exporter-app
+    catalog: default
+    clusterValues:
+      configMap: true
+      secret: false
+    forceUpgrade: true
+    namespace: kube-system
+    version: 1.8.0
```
Comment (tfussell) [inline helm/default-apps-openstack/values.yaml:75]:
```suggestion
    version: 1.9.0
```

File: helm/default-apps-openstack/templates/apps.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ $.Values.clusterName }}-{{ $appName }}-user-values
+  namespace: {{ $.Release.Namespace }}
+data:
+  {{- .configMap | toYaml | toString | nindent 2 }}
```
Comment (pipo02mix) [inline helm/default-apps-openstack/templates/apps.yaml:66]: is indent?

**Review by pipo02mix — APPROVED** [review]
LGTM if tested

File: helm/default-apps-openstack/templates/apps.yaml (hunk 6 above)
Comment (tfussell) [reply helm/default-apps-openstack/templates/apps.yaml:66]: `nindent` is `indent` with a newline prepended. I think we could also remove the `-` from `{{-` and use plain `indent`, but this is how we usually do this so I kept it.

**Pull request merged**